hal/ia32/pci.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * PCI bus access
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <hal/hal.h>
+#include "types.h"
+
+
+#define PCI_CONFIG_ADDR 0xcf8u
+#define PCI_CONFIG_DATA 0xcfcu
+#define PCI_ENABLE      0x80000000u
+
+
+const char *hal_pciClassName(u8 classId)
+{
+	static const char *pciClassNames[] = {
+		"Undefined",
+		"Mass Storage Controller",
+		"Network Controller",
+		"Display Controller",
+		"Multimedia Controller",
+		"Memory Controller",
+		"Bridge Device",
+		"Simple Communication Controller",
+		"Base System Peripheral",
+		"Input Device Controller",
+		"Docking Station",
+		"Processor",
+		"Serial Bus Controller",
+		"Wireless Controller",
+		"Intelligent I/O Controller",
+		"Satellite Communication Controller",
+		"Encryption Controller",
+		"Data Acquisition and Signal Processing Controller",
+		"Processing Accelerator",
+		"Non-Essential Instrumentation",
+	};
+
+	if (classId < (sizeof(pciClassNames) / sizeof(pciClassNames[0]))) {
+		return pciClassNames[classId];
+	}
+	else if (classId == 0x40) {
+		return "Co-Processor";
+	}
+	else if (classId == 0xff) {
+		return "Unassigned (Vendor specific)";
+	}
+	else {
+		return NULL;
+	}
+}
```

Comment:
Perhaps it is not necessary for HAL to be able to return human-readable string? Maybe it would be better to move this to `lspci` command code